deploy/awstools/awstools.py

```diff
@@ -169,14 +174,20 @@ def launch_instances(instancetype, count, instancemarket, spotinterruptionbehavi
 
     # starting with the first subnet, keep trying until you get the instances you need
     startsubnet = 0
-    instances = []
-    while len(instances) < count:
-        if not (startsubnet < len(subnets)):
-            rootLogger.critical("we tried all subnets, but there was insufficient capacity to launch your instances")
-            rootLogger.critical("""only the following {} instances were launched""".format(len(instances)))
-            rootLogger.critical(instances)
-            return
```

Comment:
I moved this down into the `except` block (and added the timeout logic) so that I could re-`raise` the exception, but then I didn't like how the output read.  I thought it was confusing to have the stacktrace at the end and switched to just using `sys.exit(1)`.